sys/include/net/ng_tcp.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2015 Simon Brummer
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_net_tcp TCP
+ * @ingroup     net
+ * @breif       RIOT's implementation of the TCP protocol
+ *
+ * @{
+ *
+ * @file
+ * @brief       TCP interface definition
+ *
+ * @author      Simon Brummer <simon.brummer@haw-hamburg.de>
+ * @}
+ */
+
+#ifndef NG_TCP_H_
+#define NG_TCP_H_
+
+#include "byteorder.h"
+#include "net/ng_netapi.h"
+#include "net/ng_netreg.h"
+#include "net/ng_nettype.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief    Default stack size for TCP thread
+ */
+#ifndef NG_TCP_STACK_SIZE
+#define NG_TCP_STACK_SIZE   KERNEL_CONF_STACKSIZE_DEFAULT
+#endif
+
+/**
+ * @brief    Default name for TCP thread
+ */
+#ifndef NG_TCP_THREAD_NAME
+#define NG_TCP_THREAD_NAME  "_NGTCP"
+#endif
+
+/**
+ * @brief Size of a TCP header
+ */
+// TODO: Depends on Headeroptions.
+#define NG_TCP_HDR_LENGTH   (20U)
+
+/**
+ *  @brief States for the TCP State Maschine
+ */
+typedef enum {CLOSED, LISTEN, SYN_SENT, SYN_RCVD, ESTABLISHED} ng_tcp_states_t;
```

Comment:
For readability and documentation I would prefer to put each enum into a single line.